### PR TITLE
Q187

### PR DESCRIPTION
**Pull request by peinanteng**

### CODE REVIEW

File: repeatedDNASequences.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+        """
+        :type s: str
+        :rtype: List[str]
+        """
+        res = []
+        if len(s) < 10:
+            return res
+        summary = {}
+        for i in range(len(s) - 9):
+            if s[i:i+ 10] in summary and summary[s[i:i+10]] == 1:
```
Comment (nkukarl) [inline repeatedDNASequences.py]: You can define `s[i : i + 10]` as a variable inside the for loop given that it is used 5 times in the for loop.

File: repeatedDNASequences.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+                    res.append(s[i:i+10])
```
Comment (nkukarl) [inline repeatedDNASequences.py]: The indentation is weird here.

File: validSudoku.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+        m = len(board)
+        l = int(math.sqrt(m))
+        for i in range(m):
+            summary = {}
+            for j in range(m):
+               if board[i][j] != '.':
+                    summary[board[i][j]] = summary.get(board[i][j], 0) + 1
```
[the remark this answers is not in the document]
Comment (peinanteng) [reply validSudoku.py]: It only contains number and '.'

File: validSudoku.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            for i in range(m):
+                if board[i][j] != '.':
+                    summary[board[i][j]] = summary.get(board[i][j], 0) + 1
+            for key, val in summary.items():
+                if val > 1:
+                    return False
+        for squareX in range(l):
+            for squareY in range(l):
+                summary = {}
+                for i in range(squareX *l , (squareX + 1) * l):
```
[the remark this answers is not in the document]
Comment (peinanteng) [reply validSudoku.py]: Sure

File: validSudoku.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+                    for j in range(squareY*l, (squareY + 1) * l):
```
[the remark this answers is not in the document]
Comment (peinanteng) [reply validSudoku.py]: Sure

File: repeatedDNASequences.py (hunk 1 above)
Comment (peinanteng) [reply repeatedDNASequences.py]: Sure

File: repeatedDNASequences.py (hunk 2 above)
Comment (peinanteng) [reply repeatedDNASequences.py]: Sure

**Pull request merged**